data-semantics.md

```diff
@@ -0,0 +1,106 @@
+
+OpenTracing defines an API through which application instrumentation can log data to a pluggable tracer. In general, there is no guarantee made by OpenTracing about the way that data will be handled by an underlying tracer. That said, the APIs are by design open-ended: what type of data should be provided to them?
+
+A general understanding between the instrumentor and tracer developers adds great value: if certain known tag key/values are used for common application scenarios, tracers can choose to pay special attention to them. The same is true of `log`ged events, and span structure in general.
+
+As an example, consider the common case of a HTTP-based application server. The URL of an incoming request that the application is handling is often useful for diagnostics, as well as the HTTP verb and the resultant status code. An instrumentor could choose to report the URL in a tag named `URL`, or perhaps named `http.url`--either would be valid from the pure API perspective. But if the Tracer wishes to add intelligence, such as indexing on the URL value or sampling proactively for requests to a particular endpoint, it must know where to look for relevant data. In short, when tag names and other instrumentor-provided values are used consistently, the tracers on the other side of the API can employ more intelligence.
+
+The guidelines contained describe a common ground on which instrumentors and tracer authors can build beyond pure data collection. Adherence to the guidelines is optional but highly recommended for instrumetors.
+
+
+
+## Spans
+
+
+### Span Naming
+
+Spans carry tags, logs, and attributes, but they also have a top-level **operation name**. This should be a low-cardinality string value representing the type of work being done in the span. Typically, a process, framework, library, or module name is a good guideline.
+
+For instance, a request being handled by a Flask-based application might be instrumented using a WSGI middleware. In this situation, the first span operation name might be `WSGI`. If the application subsequently makes a database query, the span name might be `SQL`.
+
+
+### Span Structure
+
+TODO
+
+
+
+## Semantic Tag Use-Cases
+
+The following tags are recommended for instrumentors who are trying to represent a particular type of data. Tag names follow a general structure of namespacing; the values are determined purely on a tag-by-tag basis.
+
+Some tags mentioned below may contain values of significant size. Handling of such values is implementation-specific: it is the responsibility of the Tracer to honor, drop, or truncate these tags as appropriate. However, care should be exercised on the part of the instrumentor, as even the generation or passing of such values to the Tracer may create undesirable overhead for the application.
+
+It is not required that all specified tags be used, even if one is used.
+
+
+### HTTP Server Tags
+
+These tags are recommended for spans marking entry into a HTTP-based service.
```

Comment:
I would like to defined these in `ext.tags` and announce in the documentation that people should use the constants rather than string names below. Maybe even change documentation to refer to constants (a bit difficult due to different languages).